kong/init.lua

```diff
@@ -147,6 +151,172 @@ local function load_plugins(kong_conf, dao)
   return sorted_plugins
 end
 
+
+local function mock_plugins(ctx, phase)
+  local mock    = ctx.mock     or {}
+  local plugins = mock.plugins or {}
+  local phase   = phase        or ngx.get_phase()
+
+  local BasePlugin = require "kong.plugins.base_plugin"
+
+  for plugin, plugin_conf in plugins_iterator(singletons.loaded_plugins, phase == "rewrite" or phase == "access") do
+    if plugin.handler[phase] ~= BasePlugin[phase] then
+      table.insert(plugins, {
+        name        = plugin.handler._name,
+        version     = plugin.handler.VERSION,
+        priority    = plugin.handler.PRIORITY,
+        no_consumer = plugin.schema.no_consumer,
+        phase       = phase,
+        config      = plugin_conf,
+      })
+    end
+  end
+
+  mock.plugins = plugins
+  ctx.mock     = mock
+end
+
+
+local function mock_consumer(ctx)
+  local consumer_id
+
+  local args = arguments.load({ multipart = false })
+  if args then
+    consumer_id = args.consumer
+    if type(consumer_id) == "table" then
+      consumer_id = consumer_id[1]
+    end
+  end
+
+  if consumer_id then
+    local consumer
+    local consumer_cache_key = singletons.dao.consumers:cache_key(consumer_id)
```

Comment:
It seems like we should declare/initialize this variable closer to the below `cache:get()` call? Especially considering the below checks on `type()` and what they seem to imply. (`:cache_key()` requires its arguments to be nils or strings)